### ISSUE TITLE
Add the ability to enable NTFS folder compression on Windows

### ISSUE THREAD

**Issue by jakauppila** [chef/chef#12290]
### Describe the Enhancement:
Add ability to enable NTFS folder compression on a directory on Windows.

### Describe the Need:
Enabling compression on a logs folder for example will result in much lower actual disk utilization as it's able to compress the plaintext files contained within.

### Current Alternative
We currently utilize the following WMI call to perform the compression on a directory

```
powershell_script 'compress_logs_folder' do
  code <<-EOH
  Invoke-WmiMethod -Path "Win32_Directory.Name='#{$path}'" -Name compress | Out-Null
  EOH
  not_if "if(Get-Item -path '#{path}' | Where-Object {$PSItem.Attributes -match 'Compressed'}){$true}else{$false}"
end
```

You could also achieve this via `compact /c /s:c:\docs *`

https://serverfault.com/questions/977673/how-to-enable-ntfs-folder-compression-on-windows-server-core

### Can We Help You Implement This?:
How would you like to see this implemented? My preference would be an additional windows-only property on `directory` with something like `compression true` rather than it's own resource.

When you enable compression on a folder it will compress new files going forward, it does not perform compression on existing files.

**Comment by lamont-granquist** [comment]
Yeah I think this is probably fine to do as a windows-only property on the directory resource.

The description of course should very clearly point out that it is windows only since the documentation will wind up being displayed for all consumers of the directory resource docs.

At some point in the future if this API grows more hairs, then it might be good to break out a specific windows_directory resource which `provides :directory, platform: "windows"` similarly to how the package providers work, but certainly that's way out of scope for this change, but just something to keep in mind.  As this has been pretty stable for 10+ years though I suspect that is going to wind up being unnecessary...
